great_expectations/rule_based_profiler/data_assistant/data_assistant.py

```diff
@@ -156,14 +159,23 @@ def run(
         expectation_suite_name: Optional[str] = None,
         include_citation: bool = True,
     ) -> DataAssistantResult:
-        result: DataAssistantResult = DataAssistantResult(execution_time=0.0)
-        run_profiler_on_data(
-            data_assistant=self,
+        """
+        Run the DataAssistant as it is currently configured.
+
+        Args:
+            expectation_suite: An existing "ExpectationSuite" to update.
+            expectation_suite_name: A name for returned "ExpectationSuite".
+            include_citation: Whether or not to include the Profiler config in the metadata for "ExpectationSuite" produced by "RuleBasedProfiler"
+
+        Returns:
+            DataAssistantResult: The result object for the DataAssistant
+        """
+        data_assistant_result_cls = self._data_assistant_result_cls
```

Comment:
@NathanFarmer I see the intent, though perhaps we can achieve it differently -- leave `run()` as is (or close to the way it is now) and use the `DataAssistantResult` class hierarchy to provide data to the classes, containing the charting methods we need.  Thanks.

Reply:
✅ Taken care of offline